exercises/binary-search/examples/src/BinarySearch.purs

```diff
@@ -0,0 +1,19 @@
+module BinarySearch
+  ( find
+  ) where
+
+import Prelude
+import Data.Array (length, (!!))
+import Data.Maybe (Maybe(Just, Nothing))
+
+find :: Int -> Array Int -> Maybe Int
+find i a = go 0 (length a - 1)
+  where go l h | l > h = Nothing
+        go l h =
+          let m = l + (h - l) / 2
+          in case a !! m of
```

Comment:
Yes, in this case `a !! m` will always result in a `Just j`. I did it using `!!` for two reasons.

The first is I had a view that `unsafeXXX` should be avoided unless there's a good reason. Is that not the case?

The second is that I could not really find a way to write this to be more readable / shorter. `where` in PS seems to be scoped per-guard and in this case I need `m` to be available across three guards.

I've pushed a version that uses `unsafeIndex` in case you'd prefer that - let me know if you want me to make a PR. 

https://github.com/icyrockcom/xpurescript/commit/80386982a34e4a281fe8d8aaf446d000876d4c02

If you have some suggestions on how to write this better, let me know and I'll update.